### DISCUSSION TITLE
Why cocos needed own implemention of Vector ?

### DISCUSSION THREAD

**Discussion by meiry** [axmolengine/axmol#822] [Q&A] [answered]
Hey im looking at the code trying to understand it , 
although there is comment , i the CCVector.h file . do it realy need its own implemntion of vector container ?
someone konw why its there ?

**Comment by DelinWorks** [comment, marked as answer]
Its there because it stores classes inherited from the `Ref` type, which is really useful, it retains an object in memory when it's inside the vector and releases it when it goes out. simple memory management trick

**Comment by meiry** [comment]
Thanks , i see now in the code what it doing , it encapsulate the std vector and adds method implementions of its own .
